components/SpinButton.vue

```diff
@@ -1,124 +1,116 @@
+  <b-button
+    :variant="variant"
+    :disabled="disabled"
+    :size="size"
+    :tabindex="tabindex"
+    :title="buttonTitle"
+    :class="[
+      'd-flex gap-1 align-items-center',
+      noBorder && 'no-border',
+      iconlast && 'flex-row-reverse',
+    ]"
+    @click="onClick"
+  >
+    <v-icon
+      v-if="doing"
+      icon="sync"
+      :class="['fa-spin fa-fw', spinColorClass]"
     />
-  </div>
+    <v-icon v-else-if="done && doneIcon" :icon="doneIcon" :class="iconClass" />
+    <v-icon v-else-if="iconName" :class="iconClass" :icon="iconName" />
+    <span v-if="label || $slots.default">
+      <slot>{{ label }}</slot>
+    </span>
+  </b-button>
 </template>
 <script setup>
-import { ref, defineAsyncComponent } from '#imports'
-
-const ConfirmModal = defineAsyncComponent(() => import('./ConfirmModal'))
+import { ref } from '#imports'
 
 const props = defineProps({
   variant: {
     type: String,
     required: true,
   },
-  name: {
+  iconName: {
     type: String,
     required: false,
     default: null,
   },
   label: {
     type: String,
-    required: true,
+    required: false,
+    default: '',
   },
   timeout: {
     type: Number,
     required: false,
     default: 5000,
   },
-  spinclass: {
+  spinColor: {
     type: String,
     required: false,
-    default: 'text-success',
-  },
-  disabled: {
-    type: Boolean,
-    required: false,
-    default: false,
+    default: '',
   },
+  disabled: Boolean,
   size: {
     type: String,
     required: false,
     default: null,
   },
-  iconlast: {
-    type: Boolean,
-    required: false,
-    default: false,
+  iconlast: Boolean,
+  iconClass: {
+    type: String,
+    default: 'fa-fw',
   },
-  confirm: {
-    type: Boolean,
-    required: false,
-    default: false,
+  tabindex: {
+    type: Number,
+    default: 0,
   },
-  handlerData: {
-    type: Object,
-    required: false,
-    default: null,
+  doneIcon: {
+    type: String,
+    default: 'check',
   },
-  buttonClass: {
+  buttonTitle: {
     type: String,
-    required: false,
-    default: null,
+    default: '',
   },
+  noBorder: Boolean,
 })
 
 const emit = defineEmits(['handle'])
 
 const doing = ref(false)
 const done = ref(false)
-const showConfirm = ref(false)
 
-const click = () => {
-  if (props.confirm) {
-    showConfirm.value = true
-  } else {
-    doIt()
+const spinColorClass =
+  props.spinColor ||
+  (props.variant === 'primary' ? ref('text-white') : ref('text-black'))
+
+const finnishSpinner = () => {
+  doing.value = false
+  if (props.doneIcon) {
+    done.value = true
+    setTimeout(() => {
+      done.value = false
+    }, props.timeout)
   }
 }
 
-const doIt = async () => {
+const onClick = () => {
   if (!doing.value) {
     done.value = false
     doing.value = true
 
-    await emit('handle', props.handlerData)
-
-    doing.value = false
-    done.value = true
-    setTimeout(() => {
-      done.value = false
-    }, props.timeout)
+    emit('handle', finnishSpinner)
```

Comment:
Yeah, it would be spinning until callback is called. Adding 20 sec timer + Sentry should be enough to catch places where we forget to call the callback, on it.

Reply:
Fixed that, somehow slipped through :(

Reply:
Ok - please do that.